### DISCUSSION TITLE
Fabric handshake

### DISCUSSION THREAD

**Discussion by Th0rgal** [FabricMC/fabric-api#4263] [FAQ]
Hey,

I am looking for an in depth description of the fabric handshake protocol used for checking compatibility between server and client side mods. Is there something similar to https://wiki.vg/Minecraft_Forge_Handshake#FML2_protocol_.281.13_-_Current.29 ?

Best,
Thomas

**Comment by Linguardium** [comment]
you can handle a login/connection attempt via the ServerLoginConnectionEvents.QUERY_START and sending a query using the LoginPacketSender you get. This lets you send queries to the client to respond to in order to determine compatibility. fabric itself doesnt do a mod list comparison but does handle registry syncing

**Comment by modmuss50** [comment]
See: https://github.com/FabricMC/fabric/pull/3244 for details on how we send the packet channels during configuration.
